modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearUnlockResponse.java

```diff
@@ -0,0 +1,152 @@
+
+import java.nio.ByteBuffer;
+import org.apache.ignite.internal.managers.communication.GridIoMessageFactory;
+import org.apache.ignite.internal.processors.cache.distributed.GridDistributedUnlockRequest;
+import org.apache.ignite.internal.util.typedef.internal.S;
+import org.apache.ignite.lang.IgniteUuid;
+import org.apache.ignite.plugin.extensions.communication.MessageReader;
+import org.apache.ignite.plugin.extensions.communication.MessageWriter;
+
+/**
+ * Near cache unlock request.
+ */
+public class GridNearUnlockResponse extends GridDistributedUnlockRequest {
+    /** */
+    private static final long serialVersionUID = 0L;
+
+    /** Future ID. */
+    private IgniteUuid futId;
+
+    /** Savepoint mini future ID. */
+    private int miniId;
+
+    /**
+     * Empty constructor required by {@link GridIoMessageFactory}.
+     */
+    public GridNearUnlockResponse() {
+        // No-op.
+    }
+
+    /**
+     * @param cacheId Cache ID.
+     * @param keyCnt Key count.
+     * @param addDepInfo Deployment info flag.
+     */
+    public GridNearUnlockResponse(int cacheId, int keyCnt, boolean addDepInfo, IgniteUuid futId, int miniId) {
+        super(cacheId, keyCnt, addDepInfo);
+
+        this.futId = futId;
+        this.miniId = miniId;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean writeTo(ByteBuffer buf, MessageWriter writer) {
+        writer.setBuffer(buf);
+
+        if (!super.writeTo(buf, writer))
+            return false;
+
+        if (!writer.isHeaderWritten()) {
+            if (!writer.writeHeader(directType(), fieldsCount()))
+                return false;
+
+            writer.onHeaderWritten();
+        }
+
+        switch (writer.state()) {
+            case 9:
+                if (!writer.writeIgniteUuid("futId", futId))
+                    return false;
+
+                writer.incrementState();
+
+            case 10:
+                if (!writer.writeInt("miniId", miniId))
+                    return false;
+
+                writer.incrementState();
+
+        }
+
+        return true;
+    }
+
+    /** {@inheritDoc} */
+    @Override public boolean readFrom(ByteBuffer buf, MessageReader reader) {
+        reader.setBuffer(buf);
+
+        if (!reader.beforeMessageRead())
+            return false;
+
+        if (!super.readFrom(buf, reader))
+            return false;
+
+        switch (reader.state()) {
+            case 9:
+                futId = reader.readIgniteUuid("futId");
+
+                if (!reader.isLastRead())
+                    return false;
+
+                reader.incrementState();
+
+            case 10:
+                miniId = reader.readInt("miniId");
+
+                if (!reader.isLastRead())
+                    return false;
+
+                reader.incrementState();
+
+        }
+
+        return reader.afterMessageRead(GridNearUnlockResponse.class);
+    }
+
+    /** {@inheritDoc} */
+    @Override public short directType() {
+        return 2048;
```

Comment:
Can we use named constant here?

Reply:
Yes, but such way never used. See `GridIoMessageFactory`.